stdlib-data-structures/map.cpp

```diff
@@ -0,0 +1,35 @@
+#include <iostream>
+#include <string>
+#include <map>
+
+using namespace std;
+map <string, int> cityTemperature;
+
+int main()
+{
+    string city[] = {"Paris", "London", "Berlin", "Zurich", "Bruge"};
+    int temperature[] = {10, 5, 2, -4, 3};
+    for (int i = 0; i < 5; ++i) {
+        cityTemperature[city[i]] = temperature[i];
+    }
+
+    cityTemperature["Manchester"] = 1;
+    
+    for (auto x : cityTemperature) {
+        cout << x.first << " " << x.second << '\n';
+    }
+
+    cout << cityTemperature.size() << "\n";
+
+    cityTemperature.erase("Berlin");
+
+    for (auto x : cityTemperature) {
+        cout << x.first << " " << x.second << '\n';
+    }
+    
+    cout << cityTemperature.count("Rome") << " " << cityTemperature.count("Paris") << "\n";
```

Comment:
Try the following experiment:

```
map <int, int> myMap;
myMap[5] = 10
myMap[2] = 4
cout << myMap.size();
if (myMap.count(3) == 0) {
    cout << "3 is not in map";
}
cout << myMap.size();
if (myMap[3] == 0) {
    cout << "3 is not in map";
}
cout << myMap.size();
```
Why the size of the map at the end is different? What is the difference between the 2 if statements?